File: chapter 2 - linked list/2.2.js
const LinkedList = require('./linkedlist');
const list = new LinkedList();
// generate fake data
for (let i = 50; i >= 0; i--) {
  list.add(i);
}

let p1 = list.head,
  p2 = list.head;

for (let i = 0; i <= 8; i++) {
  if (p1 === null) process.exit(0);
  p1 = p1.next;
}
while (p1 !== null) {
  p1 = p1.next;
  p2 = p2.next;
}
console.log(p2);

File: chapter 2 - linked list/2.2.py
from node import Node
from linkedlist import LinkedList

def return_kth_element(n, pos):
    p1 = n   # slow pointer
    p2 = n  # fast pointer
    print(p1.data,p2.data)
    for i in range(pos):
        # move the slow pointer to pos
        # pos will  be at (LENGTH - pos)
        if(p1 == None):
            return
        p1 = p1.next
    while(p1 != None):
        # loop till p1 is empty
        # p2 is at required position
        p1 = p1.next
        p2 = p2.next

if __name__ == "__main__":
    linked_list = LinkedList()            
    for i in reversed(range(50)):
        linked_list.add(i)
    return_kth_element(linked_list.head, 8)        
